core-backend/sandbox/src/env.rs

```diff
@@ -219,22 +259,27 @@ impl<E: Ext + IntoExtInfo + 'static> Environment<E> for SandboxEnvironment<E> {
 
             reason.unwrap_or_else(|| TerminationReason::Trap {
                 explanation: info.trap_explanation,
-                description: self.runtime.trap.map(Into::into),
+                description: trap.map(Into::into),
             })
         } else {
             TerminationReason::Success
         };
 
-        let wasm_memory_addr = self.get_wasm_memory_begin_addr();
-
-        Ok(BackendReport {
-            termination,
-            wasm_memory_addr,
-            info,
-        })
+        let gas_amount = info.gas_amount.clone();
+        post_execution_handler(wasm_memory_addr)
+            .map(|_| BackendReport { termination, info })
+            .map_err(|e| BackendError {
+                reason: e,
+                description: None,
+                gas_amount,
+            })
     }
 
-    fn drop_env(&mut self) -> GasAmount {
-        self.runtime.ext.unset().into_gas_amount()
+    fn drop_env(self) -> GasAmount {
+        self.runtime
+            .ext
+            .take()
+            .expect("method called only once with no clones around; qed")
```

Comment:
What means "qed"?

Reply:
qed = quod erat demonstrandum

This means that the expect can never fail.